packages/app/hooks/useLimitedCourse.ts

```diff
@@ -0,0 +1,27 @@
+import { API } from "@koh/api-client";
+import { GetLimitedCourseResponse } from "@koh/common";
+import useSWR, { responseInterface } from "swr";
+
+type LimitedCourseResponse = responseInterface<GetLimitedCourseResponse, any>;
+
+interface UseLimitedCourseReturn {
+  course?: LimitedCourseResponse["data"];
+  courseError: LimitedCourseResponse["error"];
+  mutateCourse: LimitedCourseResponse["mutate"];
+}
+
+export function useLimitedCourse(cid: number): UseLimitedCourseReturn {
+  const {
+    data: course,
+    error: courseError,
+    mutate: mutateCourse,
+  } = useSWR(cid && `/api/v1/courses/limited/${cid}`, async () =>
```

Comment:
Do we need an extra hook if we can just use SWR? Do we use it beyond one component?